### PR TITLE
fix unique request ID policy

### PR DESCRIPTION
**Pull request by jhendrixMSFT**

### CODE REVIEW

File: sdk/azidentity/uuid.go [hunk 1]
```diff
@@ -21,15 +21,18 @@ const (
 type uuid [16]byte
 
 // NewUUID returns a new uuid using RFC 4122 algorithm.
-func newUUID() (u uuid) {
-	u = uuid{}
+func newUUID() (uuid, error) {
+	u := uuid{}
 	// Set all bits to randomly (or pseudo-randomly) chosen values.
-	rand.Read(u[:])
+	_, err := rand.Read(u[:])
```
Comment (jhendrixMSFT) [inline sdk/azidentity/uuid.go]: `crypt/rand/Read()` is not no-fail (unlike `math/rand/Read()`) which is why I made this change.  Do we actually need the CRNG here?

**Comment by JeffreyRichter** [comment]
No, math Rand is fine. Did I say or do something that made you think crypto?


  *   Jeffrey Richter (watch Architecting Distributed Cloud Apps<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fazure.microsoft.com%2Ftraining%2Fdistributed-cloud-apps%2F&data=02%7C01%7Cjeffreyr%40microsoft.com%7C82df4ff9e5ae46a5095908d7345e957b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637035453166233741&sdata=4GnkpFUKxlZXGXFr6oQo5GB%2FFNIbDXIZDXUij0q4AEE%3D&reserved=0>)

________________________________

File: sdk/azidentity/uuid.go (hunk 1 above)
Comment (jhendrixMSFT) [reply sdk/azidentity/uuid.go]: @JeffreyRichter there are two `uuid.go` files.  The one in `azidentity` is used when creating JWTs and is the one that uses the CRNG.  I don't know if this was by accident or a requirement.  The `uuid.go` in `azcore` is using `math/rand`.

**Comment by JeffreyRichter** [comment]
All should be math; not crypto.


  *   Jeffrey Richter (watch Architecting Distributed Cloud Apps<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fazure.microsoft.com%2Ftraining%2Fdistributed-cloud-apps%2F&data=02%7C01%7Cjeffreyr%40microsoft.com%7C82df4ff9e5ae46a5095908d7345e957b%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637035453166233741&sdata=4GnkpFUKxlZXGXFr6oQo5GB%2FFNIbDXIZDXUij0q4AEE%3D&reserved=0>)

________________________________

**Comment by jhendrixMSFT** [comment]
Switched `azidentity` to use `math/rand`

File: sdk/azidentity/uuid.go [hunk 2]
```diff
@@ -4,8 +4,8 @@
 package azidentity
 
 import (
-	"crypto/rand"
 	"fmt"
+	"math/rand"
```
Comment (catalinaperalta) [inline sdk/azidentity/uuid.go]: I'm not sure I understood why we switched this?

Comment (catalinaperalta) [reply sdk/azidentity/uuid.go]: Nevermind, just saw the resolved conversation. Closing this now.

**Pull request merged**